disputer/disputer.js

```diff
@@ -70,39 +57,54 @@ class Disputer {
         Logger.error({
           at: "Disputer",
           message: "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute.",
-          id: disputeableLiquidation.id,
-          sponsor: disputeableLiquidation.sponsor
+          sponsor: disputeableLiquidation.sponsor,
+          id: disputeableLiquidation.id
         });
-
         Logger.debug({
           at: "Disputer",
-          message: "Dispute call error message",
-          id: disputeableLiquidation.id,
-          sponsor: disputeableLiquidation.sponsor,
-          error: error
+          message: `Dispute call error message: ${error.message}`,
+          error
         });
         continue;
       }
 
-      // TODO: handle transaction failures.
-      const receipt = await dispute.send({
+      Logger.info({
+        at: "Disputer",
+        message: "Disputing liquidation🔥",
+        sponsor: disputeableLiquidation.sponsor,
+        id: disputeableLiquidation.id,
+        inputPrice: priceFunction(disputeableLiquidation.liquidationTime),
         from: this.account,
         gas: 1500000,
         gasPrice: this.gasEstimator.getCurrentFastPrice()
       });
 
-      const disputeResult = {
-        tx: receipt.transactionHash,
-        sponsor: receipt.events.LiquidationDisputed.returnValues.sponsor,
-        liquidator: receipt.events.LiquidationDisputed.returnValues.liquidator,
-        id: receipt.events.LiquidationDisputed.returnValues.disputeId,
-        disputeBondPaid: receipt.events.LiquidationDisputed.returnValues.disputeBondAmount
-      };
-      Logger.info({
-        at: "Disputer",
-        message: "Dispute tx result📄",
-        disputeResult: disputeResult
-      });
+      // Send the transaction or report failure.
+      try {
+        const receipt = await dispute.send({
+          from: this.account,
+          gas: 1500000,
+          gasPrice: this.gasEstimator.getCurrentFastPrice()
+        });
+        const logResult = {
+          tx: receipt.transactionHash,
+          sponsor: receipt.events.LiquidationDisputed.returnValues.sponsor,
+          liquidator: receipt.events.LiquidationDisputed.returnValues.liquidator,
+          id: receipt.events.LiquidationDisputed.returnValues.disputeId,
+          disputeBondPaid: receipt.events.LiquidationDisputed.returnValues.disputeBondAmount
+        };
+        Logger.info({
+          at: "Disputer",
+          message: "Dispute tx result📄",
+          disputeResult: logResult
+        });
+      } catch (error) {
```

Comment:
nit: maybe we should only surround the `dispute.send` with the try catch, since we probably don't want to catch other errors.

Reply:
that's a great idea, will fix

Reply:
done